src/openrct2/ride/Ride.cpp

```diff
@@ -4346,7 +4346,8 @@ static void vehicle_create_trains(ride_id_t rideIndex, const CoordsXYZ& trainsPo
         {
             if (ride->vehicles[i] == SPRITE_INDEX_NULL)
             {
-                ride->vehicles[i] = train.head->sprite_index;
+                if (train.head != nullptr)
```

Comment:
we may as well validate on L4332 that the train has been created as there is no point validating any further beyond that point.

Reply:
So just check for train.head and train.tail and `continue` if one of them is a nullptr?

Reply:
yes